llvm/lib/Transforms/Scalar/LoopInterchange.cpp

```diff
@@ -156,7 +158,13 @@ static bool populateDependencyMatrix(CharMatrix &DepMatrix, unsigned Level,
           Dep.push_back('I');
         }
 
-        DepMatrix.push_back(Dep);
+        // Make sure we only add unique entries to the dependency matrix.
+        std::string DepStr = std::string(Dep.begin(), Dep.end());
+        if (!Seen.count(DepStr)) {
+          Seen.insert(DepStr);
```

Comment:
You should directly call insert and check the return value. That saves one hash table lookup.